NDTensors/src/lib/DiagonalArrays/src/diagonalarray/adapt.jl

```diff
@@ -0,0 +1,17 @@
+using Adapt: Adapt, adapt_storage
+
+## TODO for some reason this doesn't work in general need the specialized second function because\
+## I recieve an ambiguous function error. Looking more into this
+function Adapt.adapt_storage(to::Type{<:AbstractArray}, x::AbstractDiagonalArray)
```

Comment:
I think how you are supposed to define these is just `Adapt.adapt_storage(to, x::AbstractDiagonalArray)` without any type annotations on `to` since it could be some arbitrary adaptor.